core/contracts/Voting.sol

```diff
@@ -148,7 +173,12 @@ contract Voting is Testable, MultiRole, OracleInterface {
             "This (time, identifier) pair is not being voted on this round");
 
         VoteInstance storage voteInstance = priceResolution.votes[currentRoundId];
-        voteInstance.committedHashes[msg.sender] = hash;
+        voteInstance.voteSubmissions[msg.sender].committedHash = hash;
+
+        if (votersLastRound[msg.sender] != currentRoundId) {
+            retrieveRewards();
+            votersLastRound[msg.sender] = currentRoundId;
```

Comment:
Seems like this should be set after the reveal, not after the commit.

Reply:
Why do you say that?

Reply:
Nevermind, I was unsure if this handled the case where a voter commit but doesn't reveal during a round, but looks like its handled.